src/lib/operators/table_scan.cpp

```diff
@@ -0,0 +1,290 @@
+#include <memory>
+#include <optional>
+#include <string>
+#include <vector>
+
+#include "abstract_operator.hpp"
+#include "all_type_variant.hpp"
+#include "resolve_type.hpp"
+#include "storage/dictionary_segment.hpp"
+#include "storage/reference_segment.hpp"
+#include "storage/value_segment.hpp"
+#include "table_scan.hpp"
+#include "type_cast.hpp"
+#include "types.hpp"
+#include "utils/assert.hpp"
+
+namespace opossum {
+
+TableScan::TableScan(const std::shared_ptr<const AbstractOperator>& in, const ColumnID column_id,
+                     const ScanType scan_type, const AllTypeVariant search_value)
+    : _in{in}, _column_id{column_id}, _scan_type{scan_type}, _search_value{search_value} {}
+
+ColumnID TableScan::column_id() const { return _column_id; }
+
+ScanType TableScan::scan_type() const { return _scan_type; }
+
+const AllTypeVariant& TableScan::search_value() const { return _search_value; }
+
+std::shared_ptr<const Table> TableScan::_on_execute() {
+  auto in_table_ptr = _in->get_output();
+  auto n_chunks = in_table_ptr->chunk_count();
+  auto data_type = in_table_ptr->column_type(_column_id);
+
+  // accumulate the scan results here. We will have one chunk
+  // for each chunk in the input table (that has at least one
+  // row matching the filter).
+  auto result_chunks_ptr = std::make_shared<std::vector<std::shared_ptr<Chunk>>>();
```

Comment:
You can preserve the vector as you already have a pretty good guess for the size.

Reply:
Why do you use a shared_ptr here? It's not shared in any way and you need to dereference it once you pass it to the Table constructor.